selfdrive/manager.py

```diff
@@ -97,20 +105,30 @@ def unblock_stdout():
         line = line.rstrip()
         prefix = b'progress: '
         if line.startswith(prefix):
-          i = int(line[len(prefix):])
+          progress = 70.0 * (int(line[len(prefix):]) / TOTAL_SCONS_NODES)
           if spinner is not None:
-            spinner.update("%d" % (50.0 * (i / TOTAL_SCONS_NODES)))
-        elif len(line):
-          print(line.decode('utf8'))
+            spinner.update("%d" % progress)
+
+        if len(line) and not line.startswith(prefix):
+          line = line.decode('utf8')
+          print(line)
+          if any([err in line for err in ['error: ', 'not found, needed by target']]):
+            build_error = True
+            for _ in range(10):
```

Comment:
Why not a time.sleep(10)?

Reply:
Since I experienced the bug where it wouldn't show the error if the last spinner update is sent almost immediately for the prepare loop I wanted to ensure that never happened for the scons error. Though I never experienced it for this loop so it was just me adding it just in case it happens to happen. I can change to a time.sleep(10) if you think that it will be good

Reply:
Can you figure out why this happens? This should now be solved with a hacky sleep.

Reply:
Sure, I'll check it out tomorrow